shell/platform/windows/client_wrapper/include/flutter/plugin_registrar_windows.h

```diff
@@ -35,9 +44,78 @@ class PluginRegistrarWindows : public PluginRegistrar {
 
   FlutterView* GetView() { return view_.get(); }
 
+  // Registers |delegate| to recieve WindowProc callbacks for the top-level
+  // window containing this Flutter instance. Returns an ID that can be used to
+  // unregister the handler.
+  //
+  // Delegates are not guaranteed to be called:
+  // - The application may choose not to delegate WindowProc calls.
+  // - If multiple plugins are registered, the first one that returns a value
+  //   from the delegate message will "win", and others will not be called.
+  //   The order of delegate calls is not defined.
```

Comment:
This seems like an unnecessary restriction that can easily be lifted if the Compare template parameter for the map below was specified. Saying either the first or last one wins ought to be a much less surprising API behavior.

Reply:
It's not that simple. Plugins on Windows are DLLs (to minimize the possibility of problematic interactions), so in practice each plugin has its own instance of this class, and changing the sort behavior here would do nothing (unless a single plugin registered multiple delegates, which is possible, but unlikely). It's actually the sort order in the engine that would matter most of the time. I could define it there, but then there are bizarre edge cases like: plugin A registers a handler, then plugin B registers a handler, then plugin A registers another handler. Without retooling the API to be more complex just to handle that case, the order of calls would actually be A1 A2 B, not A1 B A2.

I also don't really see that there's any value in such a guarantee, because in practice "first" or "last" aren't really under the control of a plugin developer. Assuming people will generally register this on plugin instantiation, that means that if we make this fully defined it devolves to the order of plugin registration. That's not under the control of a plugin developer, and in practice we don't make any guarantees about that. And even if we did (which I actually filed a bug about, because it's annoying if generated code that's checked in changes for no good reason, which happened recently), then we're basically saying that WindowProc handling is decided by alphabetical ordering of plugin names, which is weird, and could have the effect of people picking odd names for plugins just to try to win conflicts.

Assuming anyone actually tested the interaction between their plugin and other plugins, which I think is going to be very rare at best.

**tl;dr**: It's non-trivial added complexity, and I don't think it has any real value.